heliopy/data/util.py

```diff
@@ -177,6 +185,43 @@ def units_attach(data, units):
     return timeseries_data
 
 
+def cdf_units(cdf_, keys=None):
+    """
+    Takes the CDF File and the required keys, and finds the units of the
+    selected keys.
+
+    Parameters
+    ----------
+    cdf_ : cdf
+        Opened cdf file
+    keys : dict, optional
+        If the user knows the units they wish to extract
+        from the CDF file keys, keys can be passed as an arugment.
+        If not, the function extracts all the keys present which
+        have an UNIT attribute.
+
+    Returns
+    -------
+    out : :class:`collections.OrderedDict`
+        Returns an OrderedDict with units of the selected keys.
+    """
+    units = coll.OrderedDict()
+    if keys is None:
+        keys = dict(zip(list(cdf_.keys()), list(cdf_.keys())))
+    for key, val in keys.items():
+        try:
+            temp_ = u.Unit(cdf_[key].attrs['UNITS'])
+        except ValueError:
+            temp_ = u.dimensionless_unscaled
```

Comment:
Can you add a warning here that dimensionless units are being automatically added

Reply:
Oh yep, you're right, totally forgot.